src/OrleansRuntime/Placement/DeploymentLoadPublisher.cs

```diff
@@ -178,6 +178,11 @@ private void NotifyAllStatisticsChangeEventsSubscribers(SiloAddress silo, SiloRu
 
         public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
         {
+            this.ScheduleTask(() => { Utils.SafeExecute(() => this.OnSiloStatusChange(updatedSilo, status), this.logger); }).Ignore();
```

Comment:
Ohh, now maybe I see.

Absolutely, definitely the original goal was to talk to system targets as grain - via grain ref, and not directly like here.
I am not sure about 1 vs 40. I am very doubtful it is indeed only 1. But it should be, at least according to the original vision, the vast majority.
There should be only 2 cases really:
1) init in silo (since system target or parts of the runtime may not exist yet, so we cant rely on regular messaging to work yet)
2) direct calls from membership, to ensure strict order under lock.

All the others can/should be replaced to grain calls to system targets. There may be some more exception, but I still think the rule is to use GR as much as possible.